third_party/pcsc-lite/naclport/server_clients_management/src/permissions_checking/user-prompting-checker.js

```diff
@@ -192,6 +181,36 @@ UserPromptingChecker.prototype.localStorageLoadedCallback_ = function(items) {
   this.localStoragePromiseResolver_.resolve(storedUserSelections);
 };
 
+/**
+ * @param {!Object} localStorageItems
+ * @return {!Map.<string, boolean>}
+ * @private
+ */
+UserPromptingChecker.prototype.parseLocalStorageUserSelections_ = function(
+    localStorageItems) {
+  /** @type {!Map.<string, boolean>} */
+  var storedUserSelections = new Map;
+  if (!localStorageItems[LOCAL_STORAGE_KEY])
+    return storedUserSelections;
+  var contents = localStorageItems[LOCAL_STORAGE_KEY];
+  if (!goog.isObject(contents)) {
+    this.logger.warning(
+        'Corrupted local storage data - expected an object, received: ' +
+        GSC.DebugDump.dump(contents));
+    return storedUserSelections;
+  }
+  goog.object.forEach(contents, function(userSelection, appId) {
+    if (!goog.isBoolean(userSelection)) {
+      this.logger.warning(
```

Comment:
Should we make some attempt at removing these from the storage?
(+) We would not spam the logs with warnings forever
(-) Harder to diagnose issues with this if we just remove stuff and warn only once.

I can't decide :-)

Reply:
Thanks for raising this question.

My primary intention was to avoid complicating the implementation because of that cleanup which wouldn't affect anything user-visible.

Also note that the bad data would be overwritten anyway by the next write operation, which will happen after the user grants a permission for some app. I'm expecting this to happen soon after our update, as, presumable, many users keep their existing set of installed extensions, which implies that a permission prompt will be shown again for the previously corrupted entries.

So I'm inclined to keep having no explicit cleanup in the code.